File: BOJ/25000 ~ 26000/P25304.java
// https://www.acmicpc.net/problem/25304
import java.util.*;
import java.io.*;

public class P25304 {
    public static void main(String[] args) throws IOException {
        InputStreamReader isr = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(isr);
        StringTokenizer st = new StringTokenizer(br.readLine());

        long x = Long.parseLong((st.nextToken())); // 영수증 총 금액

        st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()); // 물건의 갯수

        long price = 0L;
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            long a = Long.parseLong(st.nextToken());
            long b = Long.parseLong(st.nextToken());

            price += a * b;
        }
        System.out.println(x == price ? "Yes" : "No");
    }
}

File: BOJ/25000 ~ 26000/P25304.kt
// https://www.acmicpc.net/problem/25304
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

// 2. BufferedReader 사용
fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    var st = StringTokenizer(readLine())

    val x: Long = st.nextToken().toLong()
    st = StringTokenizer(readLine())
    val n: Int = st.nextToken().toInt()

    val price = IntRange(1, n).map {
        st = StringTokenizer(readLine())
        val a = st.nextToken().toLong()
        val b = st.nextToken().toLong()
        a * b
    }.reduce { acc, item -> acc + item }

    print(if (x == price) "Yes" else "No")
}

// 1. Scanner 사용 버전
//import java.util.*
//
//fun main(args: Array<String>) = with(Scanner(System.`in`)) {
//
//    val x: Long = nextLine().toLong()
//    val n: Int = nextLine().toInt()
//
//    val price = IntRange(1, n).map {
//        val (a, b) = nextLine().split(" ").map { it.toLong() }
//        a * b
//    }.reduce { acc, item -> acc + item }
//
//    print(if (x == price) "Yes" else "No")
//}
